File: java/2874.java
class Solution {
    public long maximumTripletValue(int[] nums) {
        long res = 0, diff = 0;
        int i = 0;
        for (int j : nums) {
            res = Math.max(res, diff * j);
            diff = Math.max(diff, i - j);
            if (j > i)
                i = j;
        }
        return res;
    }
}

File: dart/2874.dart
class Solution {
    int maximumTripletValue(List<int> nums) {
        int i = 0, res = 0, diff = 0;
        for (int j in nums) {
            res = max(res, diff * j);
            diff = max(diff, i - j);
            if (j > i)
                i = j;
        }
        return res;
    }
}
